### PR TITLE
Updating servers object with protocol/ serviceName

### PR DESCRIPTION
**Pull request by RoshniNaveenaS**
Fix for Github issue - https://github.tools.sap/cap/issues/issues/14440

### CODE REVIEW

File: lib/compile/csdl2openapi.js [hunk 1]
```diff
@@ -453,12 +454,16 @@ module.exports.csdl2openapi = function (
...
      * @return {Array} The list of servers
      */
-    function getServers(serviceRoot, serversObject) {
+    function getServers(serviceRoot, serversObject, csdl) {
         let servers;
 
         if (serversObject) {
             try {
                 servers = JSON.parse(serversObject);
+                // if csdl.$Version is 4.01 then protocol is rest if it is less than 4.01 then protocol is odata
```
Comment (jeevitha011) [inline lib/compile/csdl2openapi.js:463]: Can we add/update tests for this?

Comment (RoshniNaveenaS) [reply lib/compile/csdl2openapi.js:463]: added a test case to cover this scenario

File: test/lib/compile/openapi.test.js [hunk 2]
```diff
@@ -182,6 +182,17 @@ describe('OpenAPI export', () => {
...
   });
 
+  test('options: Multiple servers', () => {
+    const csn = cds.compile.to.csn(`
+      service A {entity E { key ID : UUID; };};`
+    );
+    const serverObj = "[{\n \"url\": \"https://{customer1Id}.saas-app.com:{port}/v2\",\n \"variables\": {\n \"customer1Id\": \"demo\",\n \"description\": \"Customer1 ID assigned by the service provider\"\n }\n}, {\n \"url\": \"https://{customer2Id}.saas-app.com:{port}/v2\",\n \"variables\": {\n \"customer2Id\": \"demo\",\n \"description\": \"Customer2 ID assigned by the service provider\"\n }\n}]"
+    const openapi = toOpenApi(csn, { 'openapi:servers': serverObj });
+    expect(openapi.servers).toBeTruthy();
+    expect(openapi.servers[0].url).toMatch('https://{customer1Id}.saas-app.com:{port}/v2/odata/v4A.EntityContainer')
```
Comment (jeevitha011) [inline test/lib/compile/openapi.test.js:192]: shouldn't there be a slash after odata/v4 like `https://{customer1Id}.saas-app.com:{port}/v2/odata/v4/A.EntityContainer`

[the remark this answers is not in the document]
Comment (RoshniNaveenaS) [reply test/lib/compile/openapi.test.js]: Made the changes

File: lib/compile/csdl2openapi.js [hunk 3]
```diff
@@ -453,14 +453,25 @@ module.exports.csdl2openapi = function (
...
         let servers;
 
         if (serversObject) {
             try {
-                servers = JSON.parse(serversObject);
+              servers = JSON.parse(serversObject);
+              // if csdl.$Version is 4.01 then protocol is rest if it is less than 4.01 then protocol is odata
+              const protocol = csdl.$Version <= "4.01" ? "odata/v4" : "rest";
+              // append /protocol/{$serviceName} to the URL
+              let serviceName = csdl.$EntityContainer.includes(
```
Comment (jeevitha011) [inline lib/compile/csdl2openapi.js:465]: Can we recheck this? Editing `csdl.$EntityContainer` is not ideal. I'm also not sure why we are checking for `.EntityContainer`.

Comment (RoshniNaveenaS) [reply lib/compile/csdl2openapi.js:465]: we have to get the service name which is stored here

Comment (jeevitha011) [reply lib/compile/csdl2openapi.js:465]: yes, but how it is done is not right

Comment (RoshniNaveenaS) [reply lib/compile/csdl2openapi.js]: made changes, now we take the service name like how we do for other funtions

**Review by jeevitha011 — APPROVED** [review]

**Pull request merged**